### PR TITLE
Deploy to either staging or production environments

### PR DESCRIPTION
**Pull request by emileswarts**
Previously this was hardcoded in the Makefile.
Allow deploying to different functions based on ENV environment variable

Instead of `make deploy`, it is now `make staging deploy or `make
production deploy`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,6 +4,21 @@ ifndef JENKINS_URL
   PIP_CMD = docker-compose run -v $(PWD)/vendor:/usr/src/app/vendor --rm gov-wifi-user-deletion pip
 endif
 
+check-env:
+	$(if ${ENV},,$(error Must pass ENV=<name>))
+
+staging:
+	$(eval export ENV=staging)
```
Comment (adrianclay) [inline Makefile]: What was the ` $(eval ... ) ` required for?  This [SO post](https://stackoverflow.com/a/23845391) suggests you need just ` export X = y `

Comment (emileswarts) [reply Makefile]: Oh hold up, missed the export part

Comment (emileswarts) [reply Makefile]: Oh no I didn't.  Error message as above comment.

Comment (adrianclay) [reply Makefile]: Odd, I've created a branch and it works on my machine:
https://github.com/alphagov/govwifi-data-deletion/blob/environment-specific-deployments-2/Makefile#L10

```shell
➜  govwifi-data-deletion git:(environment-specific-deployments-2) ✗ make staging
export ENV=staging
export USER_DELETION_FUNCTION=user_deletion_staging-london
export SESSION_DELETION_FUNCTION=session_deletion_staging-london
```

Comment (adrianclay) [reply Makefile]: Ahh yeah my bad.

**Pull request merged**